### PR TITLE
Coccinelle && coccigrep version bumps

### PR DESCRIPTION
**Pull request by evdenis**
Updates the deps for the coccinelle and bumps the version.

### CODE REVIEW

**Comment by evdenis** [comment]
bump

**Comment by XVilka** [comment]
ping?

File: dev-ml/parsexp/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="project">
+    <email>ml@gentoo.org</email>
```
Comment (juippis) [inline dev-ml/parsexp/metadata.xml]: Do you have their consent that they are willing to maintain this? Looking at the activity on bug / PR assigned to them...

You need to maintain it by yourself, along with the proxy-maint project unless you have indeed received some acknowledgment from the ml project.

File: dev-ml/parsexp/parsexp-0.11.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+inherit opam
+
+DESCRIPTION="S-expression parsing library"
+HOMEPAGE="https://github.com/janestreet/parsexp"
+SRC_URI="https://github.com/janestreet/parsexp/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~ppc ~x86"
```
Comment (juippis) [inline dev-ml/parsexp/parsexp-0.11.0.ebuild]: You can't add ppc on new ebuilds. It needs to be tested with proper arch team member.

File: dev-ml/parsexp/parsexp-0.11.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+IUSE=""
```
Comment (juippis) [inline dev-ml/parsexp/parsexp-0.11.0.ebuild]: Empty variables can be removed from the ebuild, although it's not necessary. Just if you prefer it cleaner that way.

File: dev-ml/sexplib0/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="project">
+    <email>ml@gentoo.org</email>
```
Comment (juippis) [inline dev-ml/sexplib0/metadata.xml]: Same here.

File: dev-ml/sexplib0/sexplib0-0.11.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+inherit opam
+
+DESCRIPTION="Library containing the definition of S-expressions and some base converters"
+HOMEPAGE="https://github.com/janestreet/sexplib0"
+SRC_URI="https://github.com/janestreet/sexplib0/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE=""
```
Comment (juippis) [inline dev-ml/sexplib0/sexplib0-0.11.0.ebuild]: And here.

File: dev-ml/sexplib0/sexplib0-0.11.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+RDEPEND=""
```
Comment (juippis) [inline dev-ml/sexplib0/sexplib0-0.11.0.ebuild]: And here.

File: dev-util/coccigrep/coccigrep-1.17.ebuild [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline dev-util/coccigrep/coccigrep-1.17.ebuild]: Won't 7 work here? Also please leave one empty line below EAPI.

File: dev-util/coccigrep/coccigrep-1.17.ebuild [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc emacs vim"
+
+RDEPEND="dev-util/coccinelle[python]
+	emacs? ( virtual/emacs )
+	vim? ( || ( app-editors/vim app-editors/gvim ) )"
+DEPEND="doc? ( dev-python/sphinx )
```
Comment (juippis) [inline dev-util/coccigrep/coccigrep-1.17.ebuild]: With EAPI-7, if it works, sphinx is a BDEPEND.

File: dev-util/coccigrep/coccigrep-1.17.ebuild [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+	emacs? ( virtual/emacs )"
+
+SITEFILE="50${PN}-gentoo.el"
+
+python_compile_all() {
+	use doc && emake -C doc html
+
+	if use emacs ; then
+		elisp-compile editors/*.el || die
```
Comment (juippis) [inline dev-util/coccigrep/coccigrep-1.17.ebuild]: This is a helper function and doesn't need to `die` in the ebuild.

File: dev-util/coccigrep/coccigrep-1.17.ebuild [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+python_install_all() {
+	use doc && local HTML_DOCS="doc/_build/html/."
+	distutils-r1_python_install_all
+
+	doman ${PN}.1
+
+	if use emacs ; then
+		elisp-install ${PN} editors/*.{el,elc} || die
```
Comment (juippis) [inline dev-util/coccigrep/coccigrep-1.17.ebuild]: Same here.

File: dev-util/coccigrep/coccigrep-1.17.ebuild [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+		elisp-site-file-install "${FILESDIR}"/${SITEFILE} || die
```
Comment (juippis) [inline dev-util/coccigrep/coccigrep-1.17.ebuild]: And here.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+SRC_URI="http://coccinelle.lip6.fr/distrib/${MY_P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc emacs ocaml +ocamlopt pcre python test vim-syntax"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+# ocaml enables ocaml scripting (uses findlib)
+CDEPEND=">=dev-lang/ocaml-3.12:=[ocamlopt?]
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Switch to `COMMON_DEPEND` here instead.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+	doc? (
+		virtual/latex-base
+		dev-texlive/texlive-latexextra
+		dev-texlive/texlive-fontsextra
+	)"
+
+DOCS=( authors.txt bugs.txt changes.txt credits.txt readme.txt )
+
+RESTRICT=strip
+S=${WORKDIR}/${MY_P}
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: `S="..."`

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+src_prepare() {
+	default
+
+	if use python ; then
+		# fix python install location
+		sed -e "s:\$(LIBDIR)/python:$(python_get_sitedir):" \
+			-i Makefile || die
+	fi
+
+	./autogen
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: wouldn't `autotools.eclass` be able to handle this? Append with `|| die` if not.

**Comment by juippis** [comment]
Sorry that the maintainers haven't noticed your contribution. If you're still willing to work on it, let's get it proper good and merge.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild (hunk 14 above)
Comment (evdenis) [reply dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Don't know how to handle this correctly https://github.com/coccinelle/coccinelle/blob/master/autogen

**Comment by evdenis** [comment]
I'll fix this in a couple of hours.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+src_prepare() {
+	default
+
+	if use python ; then
+		# fix python install location
+		sed -e "s:\$(LIBDIR)/python:$(python_get_sitedir):" \
+			-i Makefile || die
+	fi
+
+	sed -e '/have spatch/d' -i scripts/spatch.bash_completion
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: External commands need to be appended with `|| die` with optional message. (like the sed above)

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+	./autogen || die
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Try inheriting autotools and calling `eautoconf`or `eautoreconf` here, if it works would be great. 
https://devmanual.gentoo.org/general-concepts/autotools/index.html
https://devmanual.gentoo.org/eclass-reference/autotools.eclass/index.html

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild (hunk 15 above)
Comment (evdenis) [reply dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Done.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild (hunk 16 above)
Comment (evdenis) [reply dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Done.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+# ocaml enables ocaml scripting (uses findlib)
+COMMON_DEPEND=">=dev-lang/ocaml-3.12:=[ocamlopt?]
+	dev-ml/sexplib:=[ocamlopt(+)?]
+	dev-ml/menhir:=[ocamlopt?]
+	dev-ml/camlp4:=[ocamlopt?]
+	dev-ml/parmap:=[ocamlopt?]
+	emacs? ( virtual/emacs )
+	ocaml? ( dev-ml/findlib:= )
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: This doesnt have a subslot, you should remove `:=`

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+	virtual/pkgconfig
+	doc? (
+		virtual/latex-base
+		dev-texlive/texlive-latexextra
+		dev-texlive/texlive-fontsextra
+	)"
+
+DOCS=( authors.txt bugs.txt changes.txt credits.txt readme.txt )
+
+RESTRICT=strip
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: `"strip"` to be sure, its a string after all.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+		# fix python install location
+		sed -e "s:\$(LIBDIR)/python:$(python_get_sitedir):" \
+			-i Makefile || die
+	fi
+
+	sed -e '/have spatch/d' -i scripts/spatch.bash_completion || die
+
+	AT_M4DIR=setup eaclocal
+	eautoconf
+	cd bundles/stdcompat/stdcompat-current && eaclocal && eautoconf && eautomake
```
Comment (juippis) [inline dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Please split these commands into new lines, also you need to append the first `cd ...` with `|| die`

Rest looks good!

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild (hunk 17 above)
Comment (evdenis) [reply dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Done.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild (hunk 18 above)
Comment (evdenis) [reply dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Done.

File: dev-util/coccinelle/coccinelle-1.0.7.ebuild (hunk 19 above)
Comment (evdenis) [reply dev-util/coccinelle/coccinelle-1.0.7.ebuild]: Removed this line. Works without it.

**Comment by juippis** [comment]
```
 * Messages for package dev-ml/sexplib0-0.11.0:

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/ocaml/sexplib0/sexplib0.cmxs
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/ocaml/sexplib0/sexplib0.cmxs
```
```
 * Messages for package dev-ml/parsexp-0.11.0:

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/ocaml/parsexp/parsexp.cmxs
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/ocaml/parsexp/parsexp.cmxs
```
```
 * Messages for package dev-ml/sexplib-0.11.0:

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/ocaml/sexplib/unix/sexplib_unix.cmxs
 * /usr/lib64/ocaml/sexplib/num/sexplib_num.cmxs
 * /usr/lib64/ocaml/sexplib/sexplib.cmxs
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/ocaml/sexplib/num/sexplib_num.cmxs
 * /usr/lib64/ocaml/sexplib/sexplib.cmxs
 * /usr/lib64/ocaml/sexplib/unix/sexplib_unix.cmxs
```

Then, after installing sexplib0, I still can't install dependencies for coccinelle: 
```
Error: External library "sexplib.0" not found.
-> required by "src/jbuild (context default)"
Hint: try: jbuilder external-lib-deps --missing @install
make: *** [Makefile:5: default] Error 1
 * ERROR: dev-ml/base-0.9.4::gentoo failed (compile phase):
 *   emake failed
```
Any ideas?

**Comment by evdenis** [comment]
Unfortunately, I don't have a clue what could be wrong here. I will try to test it once again.

**Pull request closed without merging**